### PR TITLE
tools: Add list-repo-maintenance-commits.sh

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

We want a "monorepo maintenance" project thread (e.g. p9dueE-77C-p2) so managers can have something to look at to see the maintenance stuff we do. This script helps filter the commits for a week to ones that have some likelihood of being "maintenance" to make it easier to list them.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

None linkable, but see p9dueE-77C-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Run the script. Especially someone on a Mac.

### CODE REVIEW

File: tools/list-repo-maintenance-commits.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		$t->modify( -$d . " days" )->format( "Y-m-d" )
+	);
+')
+echo ''
+info "Listing commits from $DS to <$DE"
+
+# Collect renovate entries for special reporting, and record the lines that would otherwise be printed as "seen" so they don't get printed in other sections.
+RENOVATE=()
+while IFS= read -r LINE; do
+	SEEN[$(fmt <<<"$LINE")]=1
```
Comment (jeherve) [inline tools/list-repo-maintenance-commits.sh]: Running into an issue:

```
tools/list-repo-maintenance-commits.sh: line 103: * [Update wordpress monorepo](https://github.com/Automattic/jetpack/pull/30375): syntax error: operand expected (error token is "* [Update wordpress monorepo](https://github.com/Automattic/jetpack/pull/30375)")
```

Comment (anomiex) [reply tools/list-repo-maintenance-commits.sh]: Try it with 84d61db60d6f3fb916fc42bc5086a335da2b5d21

**Comment by jeherve** [comment]
Making progress; now running into a new issue:

```
tools/list-repo-maintenance-commits.sh: line 80: * [Init 12.2-a.2](https://github.com/Automattic/jetpack/pull/30410): syntax error: operand expected (error token is "* [Init 12.2-a.2](https://github.com/Automattic/jetpack/pull/30410)")
```

**Comment by anomiex** [comment]
Try it now with f81c4289f8

File: tools/list-repo-maintenance-commits.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
```
Comment (jeherve) [inline tools/list-repo-maintenance-commits.sh]: Turns out my shell wasn't using the latest version of bash installed on my machine. This was what was causing the errors earlier. When I explicitly enter a new shell with `bash`, I get no errors.

Would it make sense to change this to

```suggestion
#!/usr/bin/env bash
```

In my test, it helps my machine pick up the right version of Bash in my PATH.

Comment (anomiex) [reply tools/list-repo-maintenance-commits.sh]: Yeah, and we probably should also include the upgrade check too.

Comment (anomiex) [reply tools/list-repo-maintenance-commits.sh]: 312f982c52

**Review by jeherve — APPROVED** [review]

**Pull request merged**